src/pepr/operator/index.ts

```diff
@@ -114,3 +115,10 @@ When(a.Secret)
   .InNamespace("pepr-system")
   .WithName("uds-operator-config")
   .Reconcile(updateUDSConfig);
+
+// Watch the Kubernetes Clients Secret
+When(a.Secret)
+  .IsCreatedOrUpdated()
+  .InNamespace("keycloak")
+  .WithName("keycloak-client-secrets")
+  .Reconcile(updateKeycloakClientsSecret);
```

Comment:
This probably makes more sense as a mutation, rather than a watch/reconcile? If the function is entirely, only modifying the contents of the secret then it would be better to do this at mutate time. This ties into my other comment on timing/cycling of Keycloak - if you have the secret mutated before Keycloak mounts it you can dodge _some_ of those concerns.

Reply:
I typed too many words on the [other thread here](https://github.com/defenseunicorns/uds-core/pull/1341#discussion_r2007722289) which I think is the overarching issue here so going to resolve this thread to keep the discussion in one place there.